scipy/linalg/_decomp_cholesky_update.py

```diff
@@ -0,0 +1,104 @@
+               overwrite_z=False, check_finite=True):
+    """
+    Performs a rank-1 update or downdate of Cholesky factors.
+
+    Cholesky rank-1 update and downdate algorithm taken from [1]_ such that
+    an update results in ``D'D = R'R + zz'`` and a downdate results in ``D'D =
+    R'R - zz'``
+
+    Parameters
+    ----------
+    R : (N, N) ndarray
+        The 2D square array from which the triangular part will be used to
+        read the Cholesky factor.The remaining parts are ignored.
+    z : (N,) ndarray
+        A 1D modifier (update or downdate) array
+    mod_type: string, optional
+        The type of modification desired. A "+" indicates an update while a
+        "-" indicates a downdate. Default is "+" (update).
+    lower: bool, optional
+        Whether input array is upper or lower triangular Cholesky
+        factorization. Default is upper-triangular.
+    overwrite_R: bool, optional
+        If set to True the entries of the Cholesky factor array will be
+        modified during the computations instead of creating a new array.
+        Default is False.
+    overwrite_z : bool, optional
+        If set to True the entries of the update array will be modified during
+        the computations instead of creating a new array. Default is False.
+    check_finite : bool, optional
+        Whether to check that the input matrix contains only finite numbers.
+        Disabling may give a performance gain, but may result in problems
+        (crashes, non-termination) if the inputs do contain infinities or NaNs.
+        Default is True.
+
+    Returns
+    -------
+    D : (N, N) ndarray
+        The resulting modified Cholesky factor.
+
+    References
+    ----------
+    .. [1] DOI:10.1007/BF01933218
+    """
+
+    # Check for invalid mod_type argument
+    if mod_type not in ['+', '-']:
+        raise ValueError("'mod_type' argument should be one of ['+','-']")
+    # Copy or overwrite?
+    D = R if overwrite_R else R.copy()
+    z = z if overwrite_z else z.copy()
+    # Check finite, else force array
+    if check_finite:
+        R = numpy.asarray_chkfinite(R)
+        z = numpy.asarray_chkfinite(z)
+    else:
+        R = numpy.asarray(R)
+        z = numpy.asarray(z)
+    # Check for appropriate array dimensions
+    if len(R.shape) != 2:
+        raise ValueError("Expected 2D array to be updated.")
+    if len(z.squeeze().shape) != 1:
+        raise ValueError("Expected 1D update vector")
+    # Ensure square array
+    m, n = R.shape
+    if m != n:
+        raise ValueError("A square array is expected.")
+    # Ensure appropriately triangular array
+    is_proper_tri = {True: numpy.allclose(tril(R), R),
+                     False: numpy.allclose(triu(R), R)}[lower]
+    if not is_proper_tri:
+        raise ValueError(
+            "Expected {} triangular array.".format("lower" if lower else
+                                                   "upper"))
+    # Cleanup
```

Comment:
m and is_proper_tri are practically zero cost (order of bytes) you don't need to clear them.